osu.Game/Skinning/LegacyCountdown.cs

```diff
@@ -0,0 +1,146 @@
+                {
+                    count3.FadeIn(beatLength * 0.2f).ScaleTo(1, beatLength * 0.2f)
+                          .Delay(beatLength).FadeOut(beatLength * 0.2);
+                }
+
+                using (count2.BeginAbsoluteSequence(goTime - 2.2 * beatLength))
+                {
+                    count2.FadeIn(beatLength * 0.2f).ScaleTo(1, beatLength * 0.2f)
+                          .Delay(beatLength).FadeOut(beatLength * 0.2);
+                }
+
+                using (count1.BeginAbsoluteSequence(goTime - 1.2 * beatLength))
+                {
+                    count1.FadeIn(beatLength * 0.2f).ScaleTo(1, beatLength * 0.2f)
+                          .Delay(beatLength).FadeOut(beatLength * 0.2);
+                }
+
+                using (BeginAbsoluteSequence(goTime - 0.6 * beatLength))
+                {
+                    go.ScaleTo(1, 0.8 * beatLength)
+                      .Delay(0.4f * beatLength).FadeIn(0.2 * beatLength)
+                      .Delay(0.5f * beatLength).FadeOut(0.3f * beatLength);
+                }
+            }
+            else if (beatmap.HitObjects[0].StartTime > 6000)
+            {
+                // todo: some arrow thing? see stable code below
+                /*
+                int loopStartTime = hitObjectManager.hitObjects[0].StartTime - hitObjectManager.PreEmpt - 900;
+                int loopCount = ARROW_LOOP_COUNT + Math.Min(2, hitObjectManager.PreEmpt / 200);
+                ApplyArrowTransformations(loopCount, loopStartTime);
+                */
+            }
+        }
+
+        private void scheduleAudio(double goTime, double beatLength)
+        {
+            // the "readys" sample will only play if added to the skin, it does not have a default
+            ISample sampleReady = Skin.GetSample(new SampleInfo(@"readys"));
+
+            // if "count3" does not exist, the "count" sample will be used for all counts
+            // if "count3s" does exist, "count2s", "count1s" and "gos" will be used and there will not be a fallback to "count"
+            ISample sampleCount3 = Skin.GetSample(new SampleInfo(@"count3s"));
+            ISample sampleCount2;
+            ISample sampleCount1;
+            ISample sampleGo;
+
+            if (sampleCount3 == null)
+            {
+                ISample count = Skin.GetSample(new SampleInfo(@"count"));
+                sampleCount3 = count;
+                sampleCount2 = count;
+                sampleCount1 = count;
+                sampleGo = count;
+            }
+            else
+            {
+                sampleCount2 = Skin.GetSample(new SampleInfo(@"count2s"));
+                sampleCount1 = Skin.GetSample(new SampleInfo(@"count1s"));
+                sampleGo = Skin.GetSample(new SampleInfo(@"gos"));
+            }
+
+            double offset = Player.GameplayClockContainer.StartTime;
+
+            Scheduler.AddDelayed(() => sampleReady?.Play(), goTime - 5.9 * beatLength - offset);
```

Comment:
This kind of logic isn't going to work with rewinding (and probably not even with pausing, depending on the scheduler's clock). At very least this needs to use transforms, but that might not be the best way forward.

Reply:
Gotcha. Should I try readdressing this with transforms? If so, could you give me some guidance on where to start or point me to somewhere they're used in a similar way in the game?